cpp/basix/precompute.cpp

```diff
@@ -40,6 +40,10 @@ precompute::prepare_matrix(
   mdarray2_t permuted_matrix(dim, dim);
   std::vector<T> diag(dim);
 
+  const double rcond_matrix = math::rcond(
```

Comment:
Need a check here that the storage of `matrix` is contiguous, and check that it is square.